Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -693,6 +698,21 @@ void HTMLModelElement::setCurrentTime(double currentTime)
         m_modelPlayer->setCurrentTime(Seconds(currentTime), [&]() { });
 }
 
+bool HTMLModelElement::hasPortal() const
+{
+#if PLATFORM(VISION)
+    return !(document().settings().modelNoPortalAttributeEnabled() && hasAttributeWithoutSynchronization(HTMLNames::noportalAttr));
+#else
+    return true;
+#endif
+}
+
+void HTMLModelElement::updateHasPortal()
+{
+    if (RefPtr modelPlayer = m_modelPlayer)
+        modelPlayer->setHasPortal(hasPortal());
```

Comment:
I think `m_modelPlayer` is already a `RefPtr`. So can just do:
```
if (m_modelPlayer)
    m_modelPlayer->setHasPortal(hasPortal());
```

Reply:
Yes!

Reply:
Just saw Ryosuke's comment. I think he meant to store it in a local `RefPtr` to protect its reference. Please ignore my comment. Sorry about that!